### DISCUSSION TITLE
Y2038 compatibility

### DISCUSSION THREAD

**Discussion by swechencheng** [mono/mono#21809] [Q&A]
Hi,
Is there any version available or roadmap to be Y2038 compatible?
See [similar discussion](https://github.com/dotnet/core/discussions/9285).
Thanks in advance!
BR/Chencheng

**Comment by madewokherd** [comment]
As far as I can tell from a quick search, Mono already correctly accounts for a 64-bit time_t and shouldn't need any update besides a rebuild for armhf which is apparently where this changed. But it'd be good for someone to test that, I guess.

**Reply by swechencheng** [reply]
Thanks for reply. Is there any guide for test framework in this project if one can add relavant test?

**Reply by swechencheng** [reply]
BTW, do you know if which earliest version that starts using 64-bit time_t?

**Reply by madewokherd** [reply]
No idea how you'd test it.

I see it accounting for this at least as far back as the introduction of Mono.Posix in 2004. I'm not sure the assumption of 32-bit time_t was ever made.
